Sources/RxComposableArchitecture/Effects/Cancellation.swift

```diff
@@ -57,16 +57,21 @@ extension Effect {
                     }
             })
 
-            cancellationCancellables[id, default: []].insert(
-                cancellationDisposable
-            )
-
             return Observable.from(values)
                 .concat(subject)
                 .do(
                     onError: { _ in cancellationDisposable.dispose() },
                     onCompleted: cancellationDisposable.dispose,
-                    onSubscribed: { isCaching = false },
+                    onSubscribed: {
+                        isCaching = false
+                        cancellablesLock.sync {
+                            subject.onCompleted()
+                            disposable.dispose()
```

Comment:
in original pointfree, I can't find these 2 lines, intended @ivanbagaskara ?

Reply:
following existing code in line 50 jeff. should we erase .onCompleted & .dispose()?